### PR TITLE
fix(ci): restore svelte-check and clean up OpenAPI schemas

### PR DESCRIPTION
**Pull request by engels74**
## Summary

Follow-up to #173. Two pre-existing issues were called out as out of scope there; this PR resolves both.

1. **The `Type Checking` job has been silently skipping `svelte-check` for the entire lifetime of the workflow.** `.github/workflows/code-quality.yml` invokes `bun --cwd frontend run check`. With `--cwd` placed *before* `run`, bun treats it as an unrecognized top-level flag, prints its usage banner to stdout, and exits 0. Net effect: zero TypeScript errors are ever surfaced by CI. The same anti-pattern was live in two pre-commit hooks. Verified against the run logs of #173 (the step completed in roughly 50 ms).
2. **The OpenAPI-generated `frontend/src/lib/api/types.d.ts` declared `OAuthPinResponse | unknown` and `OAuthCheckResponse | unknown`.** Three OAuth endpoints in `backend/src/zondarr/api/oauth.py` annotated returns as `<TypedStruct> | Response[ErrorResponse]`. Litestar cannot infer a status code for the bare `Response[ErrorResponse]` arm, folds it into the 200 schema as `unknown`, and the union collapses to `unknown` downstream. The same issue affected `sync_server` in `backend/src/zondarr/api/servers.py`, which was papered over with a hand-written `SyncResult` interface and an `as SyncResult` cast at the call site.

Re-enabling `svelte-check` without addressing the schema would fail CI immediately, so both fixes ship together.

### CI / tooling

- Standardise on `bun run --cwd <dir> <script>` in `code-quality.yml`, both `.pre-commit-config.yaml` hooks (`typecheck-ts`, `tests-ts`), and the `CLAUDE.md` reference. This matches the only working invocation in the repo (`code-quality.yml` test job and `CLAUDE.md` line 59).

### Backend OAuth refactor (`api/oauth.py`)

- `create_pin` and `check_pin`: drop the inline `Response(ErrorResponse(...), status_code=HTTP_502_BAD_GATEWAY)` returns; raise `ExternalServiceError(provider, exc.message, original=exc)` instead. The existing `external_service_error_handler` (already registered in `app.py`) maps it to HTTP 502 with the same wire shape, plus a `correlation_id` (optional field, no consumer asserts on its absence).
- `test_complete_pin`: keeps its `Response[OAuthCheckResponse] | Response[ErrorResponse]` union but wraps every success arm in `Response(..., status_code=HTTP_200_OK)` so Litestar can score both arms. The 400 entry is documented via `responses=`.
- All three handlers gain a `responses=` decorator entry so the OpenAPI schema exposes the alternate status codes (502 for the first two, 400 for the third) without participating in the 200 schema.
- `finally: await flow.close()` blocks unchanged — `raise` and `return` have identical `finally` semantics, so the existing `mock_flow.close.assert_called_once()` assertions hold.

### Backend sync refactor (`api/servers.py`)

- New domain exception `MediaServerUnreachableError` in `core/exceptions.py` (error code `SERVER_UNREACHABLE`) and matching `media_server_unreachable_error_handler` in `api/errors.py` returning HTTP 503. Registered in `app.py` between `ExternalServiceError` and the catch-all `Exception` handler.
- `sync_server` now returns `Response[SyncResult]` and raises `MediaServerUnreachableError` from its `MediaClientError` handler. The pre-existing `_record_sync_run(..., status="failed", ...)` side effect runs *before* the raise, preserving the failed-sync record for the dashboard.
- Pinned `status_code=HTTP_200_OK` on the `@post` decorator. The endpoint previously emitted 201 (Litestar's default for `@post`) which did not match runtime intent — the controller returns a `SyncResult`, not a created resource.

### Frontend cleanup

- `frontend/src/lib/api/types.d.ts` regenerated. The `| unknown` collapse is gone; `SyncResult`, `OAuthPinResponse`, and `OAuthCheckResponse` are now typed precisely.
- `frontend/src/lib/api/client.ts`: replaced the hand-written `SyncResult` interface (and its leading TODO comment) with `export type SyncResult = components['schemas']['SyncResult']`.
- `frontend/src/routes/(admin)/servers/[id]/+page.svelte`: dropped the now-redundant `as SyncResult` cast.

### Tests

- `backend/tests/test_oauth_controller_errors.py`: the test fixture builds a Litestar instance directly without registering exception handlers; with the inline `Response` removed, raised `ExternalServiceError` would have surfaced as 500. Registered `ExternalServiceError` and `NotFoundError` handlers in `_make_test_app` so the four 502 / `flow_closed_on_error` assertions hold.
- `backend/tests/test_server_sync_status.py`: updated the manual sync integration test from `assert response.status_code == 201` to `200` to match the new explicit `status_code=HTTP_200_OK`.

## Verification

Local results on a fresh checkout of this branch:

| Gate | Result |
|------|--------|
| `bun run --cwd frontend check` | 0 errors, 0 warnings, 5083 files |
| `cd backend && uv run basedpyright` | 0 errors, 0 warnings, 0 notes |
| `cd backend && uv run pytest` | 756 passed |
| `bun run --cwd frontend test` | 302 passed (31 files) |
| `prek run --all-files --hook-stage pre-push` | All hooks pass, including `TypeScript Check` (now actually invoking `svelte-check`) |

The `TypeScript Check` hook output now contains `svelte-kit sync && svelte-check ...` followed by `START`/`COMPLETED` lines from svelte-check itself, instead of the bun usage banner.

## Test plan

- [ ] `Type Checking` job log shows real svelte-check output (not the `Usage: bun run [flags]` banner) and takes longer than ~1 second.
- [ ] All four backend test suites green (pytest, basedpyright, ruff, alembic check).
- [ ] All frontend gates green (vitest, biome, svelte-check).
- [ ] Manual smoke: trigger the Plex OAuth join flow against an unreachable Plex URL — frontend renders the existing 502 toast unchanged.
- [ ] Manual smoke: trigger user sync against an unreachable media server — frontend renders the 503 toast unchanged.

## Risks and edge cases

- `correlation_id` is now populated on OAuth 502 responses (it was omitted by the inline path). The field is `Optional` on `ErrorResponse` and no test or frontend caller asserts on its absence — net UX improvement.
- The `error_code` strings (`EXTERNAL_SERVICE_ERROR`, `SERVER_UNREACHABLE`) are unchanged and not consumed by frontend logic (`rg "EXTERNAL_SERVICE_ERROR |SERVER_UNREACHABLE" frontend/src` returns nothing).
- The `tests-ts` pre-commit hook was *also* silently passing before this change. After the fix it actually runs vitest on push; verified locally that the suite is currently green.
- After this PR, no API endpoint in the codebase uses the `Response[Foo] | Response[ErrorResponse]` anti-pattern. The five exception handlers in `api/errors.py` continue to use `Response[ErrorResponse]` — that is the correct usage there.

### CODE REVIEW

File: backend/src/zondarr/api/oauth.py [hunk 1]
```diff
@@ -262,14 +260,20 @@ async def check_pin(
...
+        },
     )
     async def test_complete_pin(
         self,
         settings: Settings,
         state: State,
         provider: Annotated[str, Parameter(description="Provider name")],
         handle: Annotated[str, Parameter(description="Opaque PIN handle")],
-    ) -> OAuthCheckResponse | Response[ErrorResponse]:
+    ) -> Response[OAuthCheckResponse] | Response[ErrorResponse]:
```
[the remark this answers is not in the document]
Comment (engels74) [reply backend/src/zondarr/api/oauth.py]: Valid. The union return annotation `OAuthCheckResponse | Response[ErrorResponse]` caused Litestar to infer the 200 schema as `oneOf: [OAuthCheckResponse, {}]`, which `openapi-typescript` collapsed to `unknown`. Fixed — implementation differs slightly from the suggestion: I added an explicit `200: ResponseSpec(data_container=OAuthCheckResponse, ...)` entry to `responses=` so the explicit schema overrides the inferred `oneOf` (chose this over converting the 400 returns to `raise` because no existing `ZondarrError` subclass produces a 400 with the plain `ErrorResponse` shape — `ValidationError` emits `ValidationErrorResponse`). Regenerated `frontend/src/lib/api/types.d.ts`; line 2788 (now ~2783) emits `components['schemas']['OAuthCheckResponse']`.

**Comment by engels74** [comment]
augment review
